CHANGELOG.md

```diff
@@ -1,65 +1,192 @@
+- test: Clean up shallow fetch tests
+  [#15002](https://github.com/rust-lang/cargo/pull/15002)
+- test: Fix `https::self_signed_should_fail` for macos
+  [#15016](https://github.com/rust-lang/cargo/pull/15016)
+- Update to libgit2 1.9.
+  [#15018](https://github.com/rust-lang/cargo/pull/15018)
+- Update dependencies: update rust crate itertools to 0.14.0
+  [#14996](https://github.com/rust-lang/cargo/pull/14996)
+  [#14998](https://github.com/rust-lang/cargo/pull/14998)
+  [#15012](https://github.com/rust-lang/cargo/pull/15012)
+
 ## Cargo 1.85 (2025-02-20)
-[66221abd...rust-1.84.0](https://github.com/rust-lang/cargo/compare/66221abd...rust-1.84.0)
+[66221abd...rust-1.85.0](https://github.com/rust-lang/cargo/compare/66221abd...rust-1.85.0)
 
 ### Added
 
 - 🎉 Cargo now supports the 2024 edition.
   More information is available in the [edition guide](https://doc.rust-lang.org/nightly/edition-guide/rust-2024/index.html).
   [#14828](https://github.com/rust-lang/cargo/pull/14828)
+- cargo-tree: The `--depth` flag now accepts `workspace`,
+  which shows only dependencies that are members of the current workspace.
+  [#14928](https://github.com/rust-lang/cargo/pull/14928)
+- Build scripts now receive a new environment variable, `CARGO_CFG_FEATURE`,
+  which contains each activated feature of the package being built.
+  [#14902](https://github.com/rust-lang/cargo/pull/14902)
+- perf: Dependency resolution is now faster due to a more efficient hash for `ActivationsKey`
+  [#14915](https://github.com/rust-lang/cargo/pull/14915)
 
 ### Changed
 
+- ❗️ cargo-rustc: Trailing flags now have higher precedence.
+  This behavior was nightly-only since 1.83 and is now stabilized.
+  [#14900](https://github.com/rust-lang/cargo/pull/14900)
+- ❗️ Cargo now uses a cross-platform hash algorithm from `rustc-stable-hash`.
+  As a result, the hash part of paths to dependency caches
+  (e.g., `$CARGO_HOME/registry/index/index.crates.io-<hash>`) will change.
+  This will trigger re-downloads of registry indices and `.crate` tarballs,
+  as well as re-cloning of Git dependencies.
+  [#14917](https://github.com/rust-lang/cargo/pull/14917)
 - Added a future-incompatibility warning for keywords in `cfg`s in Cargo.toml
   and Cargo configuration. `cfg`s with keywords like `cfg(true)` and `cfg(false)`
   were incorrectly accepted. For backward compatibility, support for raw
   identifiers has been introduced; for example, use `cfg(r#true)` instead.
   [#14671](https://github.com/rust-lang/cargo/pull/14671)
+- Dependency resolution now provides richer error messages explaining why some versions were rejected, unmatched, or invalid.  
+  [#14897](https://github.com/rust-lang/cargo/pull/14897)
+  [#14921](https://github.com/rust-lang/cargo/pull/14921)
+  [#14923](https://github.com/rust-lang/cargo/pull/14923)
+  [#14927](https://github.com/rust-lang/cargo/pull/14927)
+- build-rs: Automatically emits `rerun-if-env-changed`
```

Comment:
does this need to be a part of "Internad"?

Reply:
isn't it kinda a user-facing change?

Reply:
Not for cargo users but for `build-rs` users

Reply:
btw other `build-rs` changes are Internal

Reply:
> Not for cargo users but for `build-rs` users

Make sense. Updated.